networkx/readwrite/edgelist.py

```diff
@@ -109,8 +109,13 @@ def generate_edgelist(G, delimiter=' ', data=True):
     --------
     write_adjlist, read_adjlist
     """
-    if data is True or data is False:
-        for e in G.edges(data=data):
+    if data is True:
```

Comment:
Looks like i wanted to make a copy of the data dictionary.  Does make_str modify in place?